internal/kafka/test/mocks/kasfleetshardsync/kas-fleetshard-sync.go

```diff
@@ -68,6 +59,28 @@ var defaultUpdateDataplaneClusterStatusFunc mockKasFleetshardSyncUpdateDataPlane
 			if _, err := privateClient.AgentClustersApi.UpdateAgentClusterStatus(ctx, cluster.ClusterID, *clusterStatusUpdateRequest); err != nil {
 				return fmt.Errorf("failed to update cluster status via agent endpoint: %v", err)
 			}
+		default:
```

Comment:
The calls to `NewAuthenticatedContextForDataPlaneCluster` and `UpdateAgentClusterStatus` are duplicated.

This could be reorganized so the UpdateAgentClusterStatus is always called at the end independently of the cluster provider type.
Then we would only have a condition instead of a switch for cluster provider ocm, and it would include the calls to ocm get addons, and if the conditions of the addons are not true then we return and  end the function call.

Reply:
Great point, thanks @miguelsorianod